service/internal/cache/global.go

```diff
@@ -0,0 +1,4 @@
+package cache
+
+// Global cache that can be shared across the whole program.
+var Global = Scoped(DefaultExpiration)
```

Comment:
I'm somehow not able to follow here.

For me cache is effectively a memory buffer that is used to temporarily store something so that we don't need to reach out over the network and consume computing resources of 3rd party every time we need some given value.

The most obvious way to scope this is via memory allocation. When you allocate it the scope begins. When you free it the scope ends. Guaranteed by the language.

My earlier proposal that used `InitCtx()` was practical way to perform above allocation with scope for a single reconciliation loop.

Since we so strongly don't want to do that, we can simulate this by using the reconciliation iteration count key from context. It has the downside that you need to mix it with the cache key and take into account every time you fetch anything from cache, but it's possible.

Downside is that language doesn't guarantee you the scope anymore.

The signature of `Get(key string)` allows use of that additional scope key in the same way as `Get(scope, key string)` but gives the freedom to allow configuration of cache scope in the place of cache creation (i.e. where it's allocated and wired) instead of sprinkling it around in every place where it's used effectively rendering earlier impossible.

When we now mix the scope parameter with `object.Accessor` interface which is not related to cache implementation in any way, I feel we just make the architecture even worse.